Description: Improve timeout handling on slow machines
Author: Petter Reinholdtsen <pere@debian.org>
Bug-Debian: http://bugs.debian.org/678260
Forwarded: no
Reviewed-By: Petter Reinholdtsen <pere@hungry.com>
Last-Update: 2014-03-28
Status:  Not sent upstream

Index: startpar-git/startpar.c
===================================================================
--- startpar-git.orig/startpar.c	2014-04-10 11:17:40.332004753 +0200
+++ startpar-git/startpar.c	2014-04-10 11:17:45.675997259 +0200
@@ -1303,7 +1303,7 @@
 #if DEBUG
       fprintf(stderr, "active = %d, maxfd = %d, last = %d\n", active, maxfd, last);
 #endif
-      if (active == 1 && last >= 0)
+      if (active == 1 && last >= 0 && num >= argc)
 	{
 	  p = prgs + last;
 	  if ((pid = waitpid(p->pid, &r, (maxfd < 0 ? 0 : WNOHANG)|WUNTRACED)) == 0)
